### PR TITLE
KOGITO-9250: Add details panel to samples

### PR DESCRIPTION
**Pull request by fantonangeli**
https://issues.redhat.com/browse/KOGITO-9250
https://issues.redhat.com/browse/KOGITO-9249

**Description**

https://www.patternfly.org/v4/extensions/quick-starts/Basic-quick-starts/fullscreen-catalog-page/?

It could also have a button to use that model (besides of having a inline action on the card)

[KOGITO-9250.webm](https://github.com/kiegroup/kie-tools/assets/17780574/a1e39e5c-c21a-414e-8f7f-669cb24160c0)

### CODE REVIEW

File: packages/serverless-logic-web-tools/src/samples/ExpandableAuthor.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  }, [author]);
+
+  return (
+    <ExpandableSection
+      toggleText={author.name}
+      onToggle={() => setIsExpanded(!isExpanded)}
+      isExpanded={isExpanded}
+      isIndented
+    >
+      {author.name && (
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/samples/ExpandableAuthor.tsx]: Consider using `<When />` from react-if package to show these "optional" parts. Same for `SamplePanelContent.tsx`.

File: packages/serverless-logic-web-tools/src/samples/ExpandableAuthor.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        <div>
+          <strong>Github:</strong>{" "}
+          <a href={author.github} target="_blank" rel="noopener noreferrer">
+            {author.github}
+          </a>
+        </div>
+      )}
+      {author.social && author.social.length > 0 && (
+        <>
+          <strong>Social:</strong>
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/samples/ExpandableAuthor.tsx]: Maybe remove this "Social:" text?

File: packages/serverless-logic-web-tools/src/samples/ExpandableAuthor.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  return (
+    <ExpandableSection
+      toggleText={author.name}
+      onToggle={() => setIsExpanded(!isExpanded)}
+      isExpanded={isExpanded}
+      isIndented
+    >
+      {author.name && (
+        <div>
+          <strong>Name:</strong> {author.name}
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/samples/ExpandableAuthor.tsx]: I think we don't need to repeat the name here.

File: packages/serverless-logic-web-tools/src/samples/SampleCard.tsx [hunk 4]
```diff
@@ -148,6 +129,18 @@ export function SampleCard(props: { sample: Sample; cover: string | undefined })
...
               </Tooltip>
             </CardBody>
+            <CardFooter>
+              <Link
+                to={{
+                  pathname: routes.sampleShowcase.path({}),
+                  search: routes.sampleShowcase.queryString({ sampleId: props.sample.sampleId }),
+                }}
+              >
+                <Button variant={ButtonVariant.tertiary} ouiaId={props.sample.sampleId + `-try-swf-sample-button`}>
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/samples/SampleCard.tsx]:
```suggestion
                <Button variant={ButtonVariant.tertiary} ouiaId={props.sample.sampleId + `-try-sample-button`}>
```
Not all samples are swf.

File: packages/serverless-logic-web-tools/src/samples/SamplePanelContent.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+              <DescriptionListDescription>{relatedToList}</DescriptionListDescription>
+            </DescriptionListGroup>
+          )}
+          {resources.length > 0 && (
+            <DescriptionListGroup>
+              <DescriptionListTerm icon={<ExternalLinkAltIcon />}>Resources</DescriptionListTerm>
+              <DescriptionListDescription>{resourceList}</DescriptionListDescription>
+            </DescriptionListGroup>
+          )}
+          {status && (
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/samples/SamplePanelContent.tsx]: IMO we don't need to show the status at the moment.

File: packages/serverless-logic-web-tools/src/samples/SamplePanelContent.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+            <DescriptionListGroup>
+              <DescriptionListTerm icon={<TachometerAltIcon />}>Status</DescriptionListTerm>
+              <DescriptionListDescription style={{ textTransform: "capitalize" }}>{status}</DescriptionListDescription>
+            </DescriptionListGroup>
+          )}
+          {type && (
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/samples/SamplePanelContent.tsx]: Type here will always be sample, so we don't need to show it.

File: packages/serverless-logic-web-tools/src/samples/ExpandableAuthor.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        </div>
+      )}
+      {author.email && (
+        <div>
+          <strong>Email:</strong>{" "}
+          <a href="mailto:${author.email}" rel="noopener noreferrer">
```
Comment (caponetto) [inline packages/serverless-logic-web-tools/src/samples/ExpandableAuthor.tsx]:
```suggestion
          <a href={`mailto:${author.email}`} rel="noopener noreferrer">
```

**Review by caponetto — APPROVED** [review]

**Review by tomasdavidorg — APPROVED** [review]
Looks good to me. 🙂 

I have reported two optional issues:

- https://issues.redhat.com/browse/KOGITO-9573 - If there are more tags the button must be clicked but I would say we  should increase the limit of visible tags if it is possible. There is quite enough space.
- https://issues.redhat.com/browse/KOGITO-9572 - In the sample catalog the serverless workflow category label has orange color but green is everywhere else.

**Comment by fantonangeli** [comment]
@tomasdavidorg,
thank you for your review.
I included the fixes for your last 2 jiras (KOGITO-9573 and KOGITO-9572) and updated the video in this PR's description.

**Review by tomasdavidorg — APPROVED** [review]
Thanks @fantonangeli for the fixes 🙂. Looks good to me.

**Review by caponetto — APPROVED** [review]

**Pull request merged**